test/integration_tests/test_optimizer_rules.py

```diff
@@ -123,3 +125,96 @@ def test_should_pushdown_without_pushdown_join_rule(self):
 
         self.assertEqual(result_without_pushdown_join_rule, result_with_rule)
         self.assertEqual(query_plan, query_plan_without_pushdown_join_rule)
+
+    @patch("eva.catalog.catalog_manager.CatalogManager.get_udf_cost_catalog_entry")
+    def test_should_reorder_predicate(self, mock):
+        def side_effect_func(name):
+            if name == "DummyMultiObjectDetector":
+                return MagicMock(cost=10)
+            else:
+                return MagicMock(cost=5)
```

Comment:
Should we have a test case for a scenario where there is no cost in the catalog? (defaults to `DEFAULT_FUNCTION_EXPRESSION_COST`)

Reply:
Done!